beacon-chain/blockchain/fork_choice.go

```diff
@@ -38,24 +38,22 @@ func (c *ChainService) updateFFGCheckPts(state *pb.BeaconState) error {
 		}
 		// If the new justified slot is a skip slot in db then we keep getting it's ancestors
 		// until we can get a block.
+		lastAvailBlkSlot := lastJustifiedSlot
```

Comment:
`lastJustifiedSlot` was getting mutated as we try to find the last saved block, this caused a problem calculating state